code/datums/components/deadchat_control.dm

```diff
@@ -63,13 +63,16 @@
 		return
 
 	if(deadchat_mode & ANARCHY_MODE)
+		if(!source || !source.ckey)
+			return
 		var/cooldown = ckey_to_cooldown[source.ckey] - world.time
 		if(cooldown > 0)
-			to_chat(source, span_warning("Your deadchat control inputs are still on cooldown for another [cooldown * 0.1] seconds."))
+			to_chat(source, span_warning("Your deadchat control inputs are still on cooldown for another [CEILING(cooldown * 0.1, 1)] second\s."))
```

Comment:
Honestly the same question still applies here, why round this up?

Reply:
Having a decimal point in  a seconds value looks like a mistake. But if you round down it'll say "0 seconds." when the value is like 0.2 seconds. Which also looks like a mistake. If it were really 0 seconds, they'd already be unlocked.

Reply:
you don't think a delay of 1.5 seconds is ok?

Reply:
Pretty sure it's just the one decimal place but it does look wrong to show that to the player, yeah.

Reply:
I'm not sure I totally agree, but alright